File: #2962.count-subarrays-where-max-element-appears-at-least-k-times.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, int k) {

        int max = *max_element(nums.begin(), nums.end());
        long long result = 0;
        for (int i = 0, j = 0; j < nums.size(); j++) {
            if (nums[j] == max) {
                k--;
            }

            while (k == 0) {
                if (nums[i++] == max) {
                    k++;
                }
            }
            result += i;
        }
        return result;
    }
};

File: #2962.count-subarrays-where-max-element-appears-at-least-k-times.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number}
 */
var countSubarrays = function(nums, k) {
    const max = Math.max(...nums);
    const freq = [];
    return nums.reduce((ans, num, i) => {
        if (num === max) {
            freq.push(i);
        }
        return ans + (freq.length >= k && freq[freq.length - k] + 1);
    }, 0);
};
